# MPXV_Scripts

## Collection of scripts used for MPXV paper

![Linux](https://img.shields.io/badge/Linux-C51A4A?style=for-the-badge&logo=linux&logoColor=black)
![Ubuntu](https://img.shields.io/badge/Ubuntu-E95420?style=for-the-badge&logo=ubuntu&logoColor=white)
![R](https://img.shields.io/badge/R-Language-%23276DC3.svg?style=for-the-badge&logo=r&logoColor=white)
![Git](https://img.shields.io/badge/git-%23F05033.svg?style=for-the-badge&logo=git&logoColor=white)
![R](https://img.shields.io/badge/Be-Safe-E5320?style=for-the-badge&logo=love&logoColor=white)

## 
  
To test run these scripts in the terminal, try:

```
To use flanking_SSR_scripts, user will require three input files i.e. SSR sequence text file, one empty fasta file and a concatenated fasta file of all genomes. Two scripts are there to fetch SSR flanking regions and also to retrieve SSR_flanks common in all 404 genomes. Below are the command lines to use both the scripts.
1. perl script_1.pl input_1.txt input_2.fa 404_genomes.fa >final   {input_2.fa is an empty file}
2. perl script_2.pl
```
## Citation

If you use these scripts in your research, please cite us as follows:

  Preeti Agarwal, Nityendra Shukla, XXX, Sahil Mahfooz*, Jitendra Narayan* **Comparative Genome Analysis Reveals Insights into Driving Forces behind Monkeypox Virus Evolution and Sheds Light on the Active Role of Trinucleotide Motif ATC** 2023

BibTex:

```
@misc{
  author={Preeti Agarwal, Nityendra Shukla, XXX, Sahil Mahfooz*, Jitendra Narayan*}
  title={Comparative Genome Analysis Reveals Insights into Driving Forces behind Monkeypox Virus Evolution and Sheds Light on the Active Role of Trinucleotide Motif ATC},
  note={Version 0.1},
  year={2023}
}
```

## Help
For more information contact Preeti Agarwal [prtgrwl8@gmail.com ](mailto:prtgrwl8@gmail.com ) with any additional questions or comments.
